control-plane/connect-inject/webhook/mesh_webhook.go

```diff
@@ -382,9 +395,17 @@ func (w *MeshWebhook) Handle(ctx context.Context, req admission.Request) admissi
 				w.Log.Error(err, "error configuring injection sidecar container", "request name", req.Name)
 				return admission.Errored(http.StatusInternalServerError, fmt.Errorf("error configuring injection sidecar container: %s", err))
 			}
-			// TODO: invert to start the Envoy sidecar container before the
-			// application container
-			pod.Spec.Containers = append(pod.Spec.Containers, envoySidecar)
+			if !lifecycleEnabled {
```

Comment:
**suggestion:** Please flip this if statement as all the other if statements are checking for it enabled; this is the only that checks if it is not enabled. Also, please add a little comment about why you are doing this.